pkg/controller/azvolume.go

```diff
@@ -476,10 +463,10 @@ func (r *ReconcileAzVolume) recoverAzVolume(ctx context.Context, recoveredAzVolu
 				var err error
 				azv := obj.(*diskv1beta1.AzVolume)
 				// add a recover annotation to the CRI so that reconciliation can be triggered for the CRI even if CRI's current state == target state
-				if azv.ObjectMeta.Annotations == nil {
-					azv.ObjectMeta.Annotations = map[string]string{}
+				if azv.Status.Annotations == nil {
+					azv.Status.Annotations = map[string]string{}
 				}
-				azv.ObjectMeta.Annotations[consts.RecoverAnnotation] = "azVolume"
+				azv.Status.Annotations[consts.RecoverAnnotation] = "azVolume"
```

Comment:
It would be worth adding `getStatusAnnotations` and `setStatusAnnotations` methods the wrap common handling of a `nil` map.